### DISCUSSION TITLE
Laser module control on LE5040

### DISCUSSION THREAD

**Discussion by 1laserghost** [arkypita/LaserGRBL#919] [Q&A]
After hitting the go button to initiate a burn project on my LE5040 machine, the laser module will sometimes move to the far right side of the cross beam and attempt to keep on going. It makes a noise as the motor meets  resistance. is there a setting to keep the laser module from moving as far to the right or some other way of finding a solution?

**Comment by gmmanonymus111** [comment]
You can enable soft- or hardlimits, which suspends any movement to make sure your steppers won't crash into the end of your machine.
See a detailed description here: [Soft limit](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration#20---soft-limits-boolean), [Hard limit](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration#21---hard-limits-boolean)
Also, make sure you set your zero position by clicking on the "globe" icon after you restart your machine, as it is possible GRBL forgots the position it was before shutting down. That could explain why your machine goes off-limit.
